evalml/pipelines/component_graph.py

```diff
@@ -120,7 +121,7 @@ def fit(self, X, y):
             y (ww.DataColumn, pd.Series): The target training data of length [n_samples]
         """
         X = infer_feature_types(X)
-        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = infer_feature_types(y)
```

Comment:
This, and the similar change in `_compute_features` are important. Otherwise, if the input provided to `fit` is not a woodwork structure, fit will fail!